src/driver/driver.rs

```diff
@@ -1,47 +1,21 @@
 //! The driver trait which all drivers will implement.
 
-use schema::Definition;
-use error::{Error, ErrorCode};
-use patch::Patch;
-use value::{Pointer, Value};
-use query::Query;
+use error::Error;
+use query::{Condition, Range, Query};
+use schema::Type;
+use value::ValueStream;
 
 pub trait Driver {
   /// Connects to a driver and returns a driver instance. After calling this
   /// the driver is ready to roll!
   fn connect(uri: &str) -> Result<Self, Error> where Self: Sized;
 
-  /// Validates an Ardite Schema Definition dependending on the driver’s
-  /// contracts with the developer. Note that the definition object will not be
-  /// associated with the driver object. This is intentional, the driver should
-  /// not be able to influence the definition without an external tool.
-  fn validate_definition(_: &Definition) -> Result<(), Error> {
-    Ok(())
-  }
-
-  /// Performs a complex query on the driver. Returns a value whose shape
-  /// matches the shape of the query.
-  fn query(&self, query: Query) -> Result<Value, Error>;
-
-  /// Applies multiple patches to the driver. If one patch fails, all other
-  /// patches must also fail. Returns a value with all of the new patched
-  /// values along with any driver generated values only.
-  fn patch(&self, patch: Vec<Patch>) -> Result<Value, Error>;
-  
-  /// Get’s a value in the driver at a specific point and returns exactly that
-  /// value.
-  fn get(&self, pointer: Pointer) -> Result<Value, Error> {
-    let query = Query::from(pointer.clone());
-    let value = try!(self.query(query));
-    match value.get(pointer) {
-      Some(value) => Ok(value),
-      None => Err(Error::new(ErrorCode::Internal, "Driver failed to return a value with the requested data.", None))
-    }
-  }
-  
-  /// Set a value at a certain point in the driver. Returns nothing.
-  fn set(&self, pointer: Pointer, value: Value) -> Result<(), Error> {
-    try!(self.patch(vec![Patch::Set(pointer, value)]));
-    Ok(())
-  }
+  // TODO: doc.
+  fn read(
+    &self,
+    type_: &Type,
```

Comment:
I don't think we should change the name just because of `_`. Also, in instances like [this](https://github.com/ardite/ardite-core/blob/feat/types/src/schema/schema.rs#L37) the spec literally calls for the name `type`.

An alternative name in this instance is `collection` and in the linked instance `schema_type`. However, I have a strong distaste for both those names.